controllers/pipeline/pipeline_adapter.go

```diff
@@ -118,7 +117,6 @@ func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 		a.logger.Error(err, "Failed to create Snapshot")
 		return controller.RequeueWithError(err)
 	}
-	go metrics.RegisterNewSnapshot()
```

Comment:
It doesn't take effect here and in next place?

Reply:
Seems both.

Reply:
Why? I think when we meet error and `return controller.RequeueWithError(err)`, the `go metrics.RegisterNewSnapshot()` will not be called, but this is expected. Otherwise, `go metrics.RegisterNewSnapshot()` will be called.

Reply:
But the number of "snapshot_attempt_concurrent_requests" keeps "0" after it's called from the testing. It's weird.

Reply:
Is is really called? I feel you may add some log to see if it is called.

Reply:
The number increased for "snapshot_attempt_concurrent_total" after changing it to use "NewGaugeVec".  So, it's called.